src/fal/node_graph.py

```diff
@@ -58,6 +58,32 @@ def _add_after_scripts(
     return graph, nodeLookup
 
 
+def _add_post_hooks(
+    model: DbtModel,
+    upstream_fal_node_unique_id: str,
+    faldbt: FalDbt,
+    graph: nx.DiGraph,
+    nodeLookup: Dict[str, FalFlowNode],
+) -> Tuple[nx.DiGraph, Dict[str, FalFlowNode]]:
+    "Add post-hooks to the graph"
+    post_hooks = (
+        FalScript(faldbt, model, path, True) for path in model.get_post_hook_paths()
+    )
+    post_hook_nodes = (
+        ScriptNode(
+            _script_id_from_path(post_hook.path, model.name, "POST_HOOK"), post_hook
+        )
+        for post_hook in post_hooks
+    )
+    for post_hook_node in post_hook_nodes:
+        graph.add_node(post_hook_node.unique_id)
```

Comment:
Would it makes sense to reduce the number of for-loops here and unroll them into a single loop since each path maps to one node in the graph?

```suggestion
    for path in model.get_post_hook_paths():
        post_hook = FalScript(faldbt, model, path, True)
        script_id = _script_id_from_path(post_hook.path, model.name, "POST_HOOK")
        post_hook_node = ScriptNode(script_id, post_hook)
        graph.add_node(post_hook_node.unique_id)
 ```

Reply:
Good catch!